frontend/src/views/CellGuide/components/CellGuideCard/components/SourceDataTable/index.tsx

```diff
@@ -66,128 +65,111 @@ const SourceDataTable = ({ cellTypeId }: Props) => {
       const bFirstOrganism = bOrganisms.at(0) ?? "";
       return aFirstOrganism.localeCompare(bFirstOrganism);
     });
-    for (const collection of sortedCollections) {
+
+    return sortedCollections.map((collection, index) => {
       const tissueNames = collection.tissue.map((tissue) => tissue.label);
       const diseaseNames = collection.disease.map((disease) => disease.label);
       const organismNames = collection.organism.map(
         (organism) => organism.label
       );
-      rows.push({
-        collection: (
-          <Link
-            key={`collection-name-${collection.collection_name}-${index}`}
-            label={collection.collection_name}
-            url={collection.collection_url}
-          />
-        ),
-        publication: collection.publication_url ? (
-          <Link
-            key={`publication-url-${collection.publication_title}-${index}`}
-            label={collection.publication_title}
-            url={`https://doi.org/${collection.publication_url}`}
-          />
+
+      const tissueContent =
```

Comment:
to reduce cognitive complexity, we create some variables to capture pieces of the final output, so they're more readable